### PR TITLE
Documentation generation

### PR DESCRIPTION
**Pull request by stevegore**
Preview link: https://qctrl-python-open-controls.readthedocs-hosted.com/_/sharing/chhr71t4cloz10apwx9qrke29

I'm getting a few docstring parse errors even though numpy docstring parsing is enabled. I would imagine these can be cleaned up progressively however.

```
/Users/steve/Source/Quantum Engineering/python-open-controls/qctrlopencontrols/__init__.py:docstring of qctrlopencontrols.convert_dds_to_qiskit_quantum_circuit:6: WARNING: Unexpected indentation.
/Users/steve/Source/Quantum Engineering/python-open-controls/qctrlopencontrols/__init__.py:docstring of qctrlopencontrols.convert_dds_to_qiskit_quantum_circuit:7: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/steve/Source/Quantum Engineering/python-open-controls/qctrlopencontrols/driven_controls/driven_control.py:docstring of qctrlopencontrols.DrivenControl.export_to_file:21: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/Users/steve/Source/Quantum Engineering/python-open-controls/qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py:docstring of qctrlopencontrols.DynamicDecouplingSequence.export_to_file:25: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/Users/steve/Source/Quantum Engineering/python-open-controls/qctrlopencontrols/driven_controls/driven_control.py:docstring of qctrlopencontrols.driven_controls.driven_control.DrivenControl.export_to_file:21: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/Users/steve/Source/Quantum Engineering/python-open-controls/qctrlopencontrols/dynamic_decoupling_sequences/dynamic_decoupling_sequence.py:docstring of qctrlopencontrols.dynamic_decoupling_sequences.dynamic_decoupling_sequence.DynamicDecouplingSequence.export_to_file:25: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/Users/steve/Source/Quantum Engineering/python-open-controls/qctrlopencontrols/qiskit/quantum_circuit.py:docstring of qctrlopencontrols.qiskit.quantum_circuit.convert_dds_to_qiskit_quantum_circuit:6: WARNING: Unexpected indentation.
/Users/steve/Source/Quantum Engineering/python-open-controls/qctrlopencontrols/qiskit/quantum_circuit.py:docstring of qctrlopencontrols.qiskit.quantum_circuit.convert_dds_to_qiskit_quantum_circuit:7: WARNING: Block quote ends without a blank line; unexpected unindent.
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,8 +99,34 @@ See the [Jupyter notebooks](https://github.com/qctrl/notebooks/tree/master/qctrl
...
+For general guidelines, see [Contributing](https://github.com/qctrl/.github/blob/master/CONTRIBUTING.md).
+
+### Building documentation
+
+Documentation generation relies on [Spinx](http://www.sphinx-doc.org). Automated builds are done by [Read The Docs](https://readthedocs.com).
+
+To build locally:
+
+1. Ensure you have used one of the install options above.
+1. Ensure you have the environment variable APITOKEN set to a valid token.
```
Comment (charmasaur) [inline README.md]: Update numbering

Comment (charmasaur) [inline README.md]: Worth clarifying what sort of token this should be?

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+# Add any paths that contain custom static files (such as style sheets) here,
+# relative to this directory. They are copied after the builtin static files,
+# so a file named "default.css" will overwrite the builtin "default.css".
+html_static_path = ['_static']
+html_css_files = [
+    'css/custom.css',
+]
+
+# token = os.environ.get("QCTRL_TEST_TOKEN")
```
Comment (charmasaur) [inline docs/conf.py]: Should this block still be here?

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,134 @@
+
```
Comment (charmasaur) [inline docs/index.rst]: Is this file auto-generated from the README? Or do we need to manually keep it up to date?

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+Anyone interested in quantum control is welcome to contribute to this project.
+
+Table of Contents
+-----------------
+
+
+* `Installation <#installation>`_
+* `Usage <#usage>`_
+* `Contributing <#contributing>`_
```
Comment (charmasaur) [inline docs/index.rst]: Doesn't look like there are "Contributing" or "Credits" sections below (but there is a "Reference")

File: pyproject.toml [hunk 5]
```diff
@@ -39,6 +39,8 @@ qiskit-terra = "^0.8.1"
 qiskit-ibmq-provider = "^0.2.2"
 cirq = "^0.5.0"
 pyquil = "^2.9"
+sphinx = "^2.2.0"
```
Comment (charmasaur) [inline pyproject.toml]: Should these be dev-dependencies?

File: README.md (hunk 1 above)
Comment (stevegore) [reply README.md]: Actually it renders fine. This was a standards discussion a while ago.

Comment (stevegore) [reply README.md]: This is a copy/paste error! Thank you

File: docs/conf.py (hunk 2 above)
Comment (stevegore) [reply docs/conf.py]: Actually I'll remove completely.

File: README.md (hunk 1 above)
Comment (charmasaur) [reply README.md]: Mind blown, thanks!

File: docs/index.rst (hunk 3 above)
Comment (stevegore) [reply docs/index.rst]: Good question. That's a concern I have too. I would expect that in all likelihood, the README will eventually just point to the documentation on RTD.

File: docs/index.rst (hunk 4 above)
Comment (stevegore) [reply docs/index.rst]: Gagh sorry. I clearly rushed this one.

**Comment by stevegore** [comment]
Hey thanks for the great review. I'll make changes after this meeting.

File: docs/index.rst (hunk 3 above)
Comment (charmasaur) [reply docs/index.rst]: OK, that seems like a good solution to me.

File: docs/conf.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+#
+import os
+import sys
+import toml
+
+sys.path.insert(0, os.path.abspath('../qctrl'))
```
Comment (rajibchakravorty) [inline docs/conf.py]: I think `os.path.abspath('../qctrl')` needs changing; For some reason, on my machine, changing this to `os.path.abspath(..)` works, while in my opinion it should have been `os.path.abspath('../qctrlopencontrols')`.

Comment (stevegore) [reply docs/conf.py]: Great pickup.

File: pyproject.toml (hunk 5 above)
Comment (stevegore) [reply pyproject.toml]: They are now dev-dependencies

**Pull request merged**